symbol_exporter/ast_package_symbol_extractor.py

```diff
@@ -0,0 +1,164 @@
+import ast
+from collections import defaultdict
+from graphlib import TopologicalSorter
+from pathlib import Path
+from typing import Union
+
+from symbol_exporter.ast_symbol_extractor import (
+    SymbolFinder,
+    is_relative_import,
+    SymbolType,
+    is_relative_star_import,
+)
+
+
+def get_symbol_type(symbol) -> SymbolType:
+    return symbol["type"]
+
+
+def is_package(directory: Path) -> bool:
+    # Only handles regular packages, not namespace packages. See PEP-420.
+    return (directory / "__init__.py").exists()
+
+
+def parse(module: Path, module_path: str) -> dict:
+    code = module.read_text()  # TODO: need to check for multiple encodings
+    tree = ast.parse(code)
+    z = SymbolFinder(module_name=f"{module_path}.{module.stem}")
+    z.visit(tree)
+    return z.post_process_symbols()
+
+
+def merge_dicts(*dicts):
+    new_symbols = {}
+    for d in dicts:
+        new_symbols |= d
+    return new_symbols
+
+
+def resolve_relative_import(module: str, level: int, shadows: str) -> str:
+    """Expects data of a relative import"""
+    parent = module.split(".")[:-level]
+    return ".".join([*parent, shadows])
+
+
+class _RelativeImportsResolver:
+    def __init__(self, symbols):
+        self._original_symbols = symbols
+        self._sorted_symbols = {}
+        self._namespaces = {}
+
+    def resolve(self) -> dict:
+        self._normalize_and_sort_symbols()
+        return self._resolve_star_imports()
+
+    def _normalize_and_sort_symbols(self):
+        """Removes __init__.py from symbols and returns dict sorted by symbol type"""
+        tmp_sorted = {}
+        topological_sorter = TopologicalSorter()
+        relative_star_imports_volume = {}
+        namespaces = defaultdict(list)
+        for k, v in sorted(self._original_symbols.items(), key=lambda x: x[1]["type"]):
+            new_symbol = k.replace(".__init__", "")
+            if is_relative_import(v):
+                shadows = resolve_relative_import(**v["data"])
+                data = dict(v, data=dict(shadows=shadows))
+                tmp_sorted[new_symbol] = data
+                namespace, symbol = new_symbol.rpartition(".")[:3:2]
```

Comment:
Would this work?
```suggestion
                namespace, symbol = new_symbol.rpartition(".")[::2]
```

I am making this comment solely in my personal capacity and am not conveying any rights to any intellectual property of any third parties.

Reply:
Or
```suggestion
                namespace, _ symbol = new_symbol.rpartition(".")
```

I am making this comment solely in my personal capacity and am not conveying any rights to any intellectual property of any third parties.

Reply:
I guess the `[:3:2]` threw me for a loop.

I am making this comment solely in my personal capacity and am not conveying any rights to any intellectual property of any third parties.

Reply:
yeah, basically I wanted to get index 0 and 2 in the array, both of these work.

The second option maybe makes it clearer what we are trying to do, which is ignore the scorn element.